service/scheduler/quartz/src/main/java/org/eclipse/kapua/service/scheduler/trigger/quartz/TriggerServiceImpl.java

```diff
@@ -87,7 +90,14 @@ public Trigger create(TriggerCreator triggerCreator)
         authorizationService.checkPermission(permissionFactory.newPermission(SCHEDULER_DOMAIN, Actions.write, triggerCreator.getScopeId()));
 
         return entityManagerSession.onTransactedInsert(em -> {
-            // Kapua Trigger definition
+
+            KapuaQuery<Trigger> triggerQuery = queryByJobId;
+            TriggerListResult list = query(triggerQuery);
+            for (Trigger trigger : list.getItems()) {
```

Comment:
This is a very inefficient way to do it. 
If someone has 5000 of triggers you'll need to scan them all, which is far more inefficient than just query by name.